math/calculator.cpp

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright 2020 @author David Leal
+ * @file
+ *
+ * Basic calculator: sum, subtract, multiply, and divide.
+ */
+
+#include <iostream>
+#include <cstdlib>
+
+/**
+ * Same as the number typed by the user, but it's saved
+ * in the new node so it won't be re-writed.
+ */
+
+struct Node {
+    float datatype;
+};
+
+/** 
+ * Prototypes
+ */
+
+void requestData();
+void pause();
+
+Node* new_node = new Node();
```

Comment:
avoid using global variables. Create them in the functions and pass as function arguments instead.

Reply:
Fixed.

Reply:
> avoid using global variables. Create them in the functions and pass as function arguments instead.
> 
> This will prevent inadvertent bugs. for example, the node here is created as a dynamically allocate memory block __and should be "deleted" otherwise, leads to memory leaks.__

Thank you. The memory allocated using `new` keyword **MUST** be freed/deleted.

Reply:
> **MUST** be freed/deleted.

Fixed, allocated memory is now released.
Also added documentation about its use.